subt/ros/proxy/ros_proxy_node.cc

```diff
@@ -602,8 +602,25 @@ void Controller::receiveZmqThread(Controller * self)
       }
       else
       {
-        ROS_INFO_STREAM("MD bad parsing" << c << " " << buffer);
-        break;
+        double sx, sy, sz;
+        double ax, ay, az;
+
+        int c = sscanf(buffer, "cmd_vel_3d %lf %lf %lf %lf %lf %lf", &sx, &sy, &sz, &ax, &ay, &az);
+        if(c == 6)
+        {
+          msg.linear.x = sx;
+          msg.linear.y = sy;
+          msg.linear.z = sz;
+          msg.angular.x = ax;
+          msg.angular.y = ay;
+          msg.angular.z = az;
+          self->velPub.publish(msg);
+        }
+        else
+        {
+          ROS_INFO_STREAM("MD bad parsing" << c << " " << buffer);
+          break;
+        }
```

Comment:
I'd rather stop extending `ros_proxy_node.cc` and moved to `rospy`-based solution. It could be part of `cloudsim2osgar.py` (zmq receive instead of `rospy.spin()`) or we can create another `osgar2cloudsim.py` node.

Reply:
I am actually not sure if to continue with this PR - what is left is this support of cmd_vel3d for both `rosmsg.py` and `ros_proxy_node.cc` which you want to remove. Dummy skeleton for OSGAR height control can be easily rewritten ... hmm.

Reply:
We now have a path from osgar to cloudsim over rospy. That could be used to send the speed commands. I am saying that I am not sure if there is anything left worth integrating.

Reply:
OK - you would have to synchronize ROS motion control some way, i.e. rosmsg vs. rospy ... also it has to be regularly updated as far as I remember ... so yes, but there could be some ugly details (as usual).